File: sdk/php/src/events/RemoveSignals.php
<?php
/**
 * @copyright Copyright (c) PutYourLightsOn
 */

namespace aereaco\nexus-ux\events;

use aereaco\nexus-ux\Consts;
use aereaco\nexus-ux\enums\EventType;

class RemoveSignals implements EventInterface
{
    use EventTrait;

    public ?array $paths;

    public function __construct(array $paths = null, array $options = [])
    {
        $this->paths = $paths;

        foreach ($options as $key => $value) {
            $this->$key = $value;
        }
    }

    /**
     * @inerhitdoc
     */
    public function getEventType(): EventType
    {
        return EventType::RemoveSignals;
    }

    /**
     * @inerhitdoc
     */
    public function getDataLines(): array
    {
        $dataLines = [];

        foreach ($this->paths as $path) {
            $dataLines[] = $this->getDataLine(Consts::PATHS_DATALINE_LITERAL, $path);
        }

        return $dataLines;
    }
}

File: sdk/java/core/src/main/java/aereaco/nexusux/RemoveSignals.java
package aereaco.nexusux;

import aereaco.nexusux.enums.EventType;

import java.util.ArrayList;
import java.util.List;

public class RemoveSignals implements Event {
    private final List<String> paths;
    private String eventId;
    private Integer retryDuration;

    public RemoveSignals(List<String> paths) {
        this.paths = paths;
    }

    @Override
    public EventType getEventType() {
        return EventType.RemoveSignals;
    }

    @Override
    public List<String> getDataLines() {
        List<String> dataLines = new ArrayList<>();
        for (String path : paths) {
            dataLines.add(String.format("data: %s %s", Consts.PATHS_DATALINE_LITERAL, path));
        }
        return dataLines;
    }

    @Override
    public String getEventId() {
        return eventId;
    }

    @Override
    public Integer getRetryDuration() {
        return retryDuration;
    }
}